src/rpc/rawtransaction.cpp

```diff
@@ -99,6 +100,19 @@ void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry)
 
             in.push_back(Pair("value", ValueFromAmount(spend->getIntDenomination())));
             in.push_back(Pair("valueSat", spend->getIntDenomination()));
+        } else if (txin.IsZerocoinRemint()){
+            std::shared_ptr<sigma::CoinRemintToV3>  remint;
+            try {
+                CDataStream serData(std::vector<unsigned char>(txin.scriptSig.begin()+1, txin.scriptSig.end()), SER_NETWORK, PROTOCOL_VERSION);
+                remint = std::make_shared<sigma::CoinRemintToV3>(serData);
+                in.push_back(Pair("publicCoinValue", remint->getPublicCoinValue().ToString(16)));
+            } catch (std::ios_base::failure &) {
+                throw JSONRPCError(RPC_DATABASE_ERROR, "An error occurred during processing the Zerocoin to Sigma remint information");
+            }
+            fillStdFields(in, txin);
+
+            in.push_back(Pair("value", remint->getDenomination()));
```

Comment:
We should format the value with `ValueFromAmount` so it will be the same format as the others.

Reply:
Sorry, I won't as ValueFromAmount converts sats to coins. I am doing it in the opposite direction.

Reply:
The problems with the current way is it will show without fractional while the others will show as fractional even it is a whole coin (e.g. 1 XZC on the others will show as `1.00000000` but this one will show just `1`).

Reply:
Aha, okey

Reply:
Done correct I think, remint->getDenomination() returns fractional value.

Reply:
For me not printing the leading zeroes looks more correct as well, as it saves electrons both on producing the output and parsing it. But since we have this approach everywhere, I thought it is better to follow it, so I added...